external/storm-jms/src/main/java/org/apache/storm/jms/spout/JmsSpout.java

```diff
@@ -127,8 +130,14 @@ public void setJmsAcknowledgeMode(final int mode) {
                 messageHandler = new TransactedSessionMessageHandler();
                 break;
             default:
-                LOG.warn("Unsupported Acknowledge mode: "
-                    + mode + " (See javax.jms.Session for valid values)");
+                // individual message ack-ing needs vendor specific mode
+                if (individualAcks) {
```

Comment:
This change requires end users to call individualAcks earlier than this method. Could we guide this in javadoc or in exception message? Or can we fix it? (Maybe it defers verification step in open() so violating fail-fast...)